src/halmos/sevm.py

```diff
@@ -2247,20 +2340,30 @@ def run(self, ex0: Exec) -> Tuple[List[Exec], Steps]:
                     ex.st.push(uint256(f_origin()))
                 elif opcode == EVM.ADDRESS:
                     ex.st.push(uint256(ex.this))
+                # TODO: define f_extcodesize for known addresses in advance
                 elif opcode == EVM.EXTCODESIZE:
-                    address = uint160(ex.st.pop())
-                    if address in ex.code:
-                        codesize = con(len(ex.code[address]))
+                    account = uint160(ex.st.pop())
+                    account_addr = self.resolve_address_alias(ex, account)
+                    if account_addr is not None:
+                        codesize = con(len(ex.code[account_addr]))
                     else:
-                        codesize = f_extcodesize(address)
+                        codesize = f_extcodesize(account)
                         if (
-                            address == hevm_cheat_code.address
-                            or address == halmos_cheat_code.address
+                            eq(account, hevm_cheat_code.address)
+                            or eq(account, halmos_cheat_code.address)
+                            or eq(account, console.address)
                         ):
                             ex.solver.add(codesize > 0)
                     ex.st.push(codesize)
+                # TODO: define f_extcodehash for known addresses in advance
                 elif opcode == EVM.EXTCODEHASH:
-                    ex.st.push(f_extcodehash(ex.st.pop()))
+                    account = uint160(ex.st.pop())
+                    account_addr = self.resolve_address_alias(ex, account)
+                    if account_addr is not None:
+                        codehash = f_extcodehash(account_addr)
+                    else:
+                        codehash = f_extcodehash(account)
```

Comment:
you know I'm a big fan of these expressions 🥰 

```suggestion
                    codehash = (
                        f_extcodehash(account_addr)
                        if account_addr is not None
                        else f_extcodehash(account)
                    )
```

Reply:
this is great btw, we're getting a ton of mileage out of the address aliases!

Reply:
um, tbh, the ite expr is not readable to me if it doesn't fit in a single line.

Reply:
readability isn't great but it's still cleaner IMO to have:
- a single assignment to codehash vs two
- the assignment be in scope of the use later